EventSelectionEfficiency/20240913_Closure/makeEvtSel.cpp

```diff
@@ -0,0 +1,863 @@
+#include <iostream>
+#include <vector>
+#include <map>
+using namespace std;
+
+// root includes
+#include "TTree.h"
+#include "TFile.h"
+#include "TStyle.h"
+#include "TH2D.h"
+#include "TCanvas.h"
+#include "TGraph.h"
+#include "TGaxis.h"
+#include "TLatex.h"
+#include "TLegend.h"
+
+#include "Messenger.h"
+#include "CommandLine.h"
+#include "Matching.h"
+#include "ProgressBar.h"
+#include "TauHelperFunctions3.h"
+#include "SetStyle.h"
+
+
+
+int main(int argc, char *argv[]);
+int FindBin(double Value, int NBins, double Bins[]); 
+void MakeCanvasZ(vector<TH1D > Histograms, vector<string> Labels, string Output, string X, string Y, double WorldMin, double WorldMax, bool DoRatio, bool LogX); 
```

Comment:
We should use vector<TH1D> &Histograms.  Otherwise whenever you call the functions, all the histograms will be recreated.  For other smaller structure (string, etc) this is fine but Histograms are kind of complicated object and it's better to pass by reference.